src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,88 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.exception.WrongValidationException;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private RegistrationService registrationService;
+    private User newUser;
+
+    @BeforeEach
+    void setUp() {
+        newUser = new User("ElonMusk99", "eLONmUSK((", 24);
```

Comment:
It's better to create a user in each method to make the tests more readable and understandable.

This is still a topic for discussion. 
Your test should consist of three blocks: given - when - then.
Creating set parameters to test helps other developers understand what parameters you are setting and testing, and avoids looking at the beginning of a test class each time they open different methods.

Also, set age 18 like the min limit. 
If someone by mistake in your code change this parameter your register_validUser_Ok test method doesn't throw the exception.

Reply:
I don`t understand the last paragraph. I have a constant with min age in RegistrationServiceImpl.java. If I set value less than this constant, I get the RegisterException. I fixed the description of this exception in case age is less this constant.